### DISCUSSION TITLE
[Feedback / Feature Request] The new UI for environment variables needs work

### DISCUSSION THREAD

**Discussion by amille14** [vercel/vercel#5570] [General] [answered]
Previously, env vars were split into separate tabs for production / preview (staging) / development. With the recent update to the env vars UI, it's just a giant list of all variables for all environments with no sorting or filtering whatsoever. This makes it very difficult to keep track of which environment has which variables set when you have many variables.

Please add the ability to:
1. Filter variables by environment so that I can see only my production env vars or only my staging env vars, for example.
2. Sort variables alphabetically so I can see similarly-named variables side by side. This makes it easier when working with multiple vars with the same prefix (such as AWS_ or GOOGLE_ or REACT_APP_, etc.).
3. Simple search by variable name. For now, ctrl + f in the browsers works alright to find a specific variable, but a basic search text field that supports partial matches would be nice, especially since variables are not sorted alphabetically at the moment.

The ability set the same variable for multiple environments was a step in the right direction, but it's still really hard to manage environment variables in multiple environments properly at the moment. I hope some of these issues will be addressed soon.

**Comment by leo** [comment, marked as answer]
Thank you for your feedback!

We just launched a [new UI](https://vercel.com/changelog/2020-12-23-environment-variables-can-now-be-filtered) for searching Environment Variables and filtering them by Environment, which covers `#1` and `#3`.

For `#2`, we don't have a solution yet, but we're exploring how to solve that the best, as we'd still like to move the ones that were added most recently to the top.
